netplugin panic while deleting vtep interface

issue:
======
netplugin panics while trying to delete vtep interface

analysis:
========
Initial analysis is that we do multiple registration for peer discovery. And
then we try and free up same resource multiple times. This causes the panic 
```
➜  gists  grep -rn "CreatePeerHost" np_274.log
np_274.log:168:time="Feb  9 06:15:59.108602544" level=info msg="CreatePeerHost for {HostAddr:192.168.2.11 Port:9002}"
np_274.log:171:time="Feb  9 06:15:59.421090603" level=info msg="CreatePeerHost for {HostAddr:192.168.2.11 Port:9010}"
➜  gists  grep -rn "DeletePeerHost" np_274.log
np_274.log:293:time="Feb  9 06:23:01.229189305" level=info msg="DeletePeerHost for {HostAddr:192.168.2.11 Port:9002}"
np_274.log:296:time="Feb  9 06:23:01.233382164" level=info msg="DeletePeerHost for {HostAddr:192.168.2.11 Port:9010}"
np_274.log:305:github.com/contiv/netplugin/drivers.(*OvsDriver).DeletePeerHost(0xc8201d8c40, 0xc8204180dd, 0xc, 0x2332, 0x0, 0x0)
np_274.log:307:github.com/contiv/netplugin/netplugin/plugin.(*NetPlugin).DeletePeerHost(0xc82005bb00, 0xc8204180dd, 0xc, 0x2332, 0x0, 0x0)
➜  gists
```

steps:
======
```
vagrant@netplugin-node1:~$ netctl net create contiv-net --subnet=20.1.1.0/24 --gateway=20.1.1.254 --pkt-tag=1001
vagrant@netplugin-node1:~$ docker run -itd --name=web --net=contiv-net ubuntu /bin/bash
Unable to find image 'ubuntu:latest' locally
latest: Pulling from library/ubuntu
92ec6d044cb3: Pull complete
2ef91804894a: Pull complete
f80999a1f330: Pull complete
6cc0fc2a5ee3: Pull complete
Digest: sha256:457b05828bdb5dcc044d93d042863fba3f2158ae249a6db5ae3934307c757c54 Status: Downloaded newer image for ubuntu:latest
eee20636ec1f6a077a5766b22a5b770dd2464058e8d1660f0ba9f5a45e328007
Error response from daemon: Cannot start container eee20636ec1f6a077a5766b22a5b770dd2464058e8d1660f0ba9f5a45e328007: Post http:///run/docker/plugins/netplugin.sock/IpamDriver.RequestAddress: dial unix/run/docker/plugins/netplugin.sock: connection refused
vagrant@netplugin-node1:~$
```

panic trace:
=========
```
panic: runtime error: index out of range

goroutine 147 [running]:
github.com/contiv/netplugin/drivers.(*OvsdbDriver).GetOfpPortNo(0xc8201686a0, 0xc8205c58b0, 0xd, 0x2, 0x0, 0x0)
	/opt/gopath/src/github.com/contiv/netplugin/drivers/ovsdbDriver.go:578 +0x86f
github.com/contiv/netplugin/drivers.(*OvsSwitch).DeleteVtep(0xc820164a20, 0xc8204180dd, 0xc, 0x0, 0x0)
	/opt/gopath/src/github.com/contiv/netplugin/drivers/ovsSwitch.go:458 +0x1f7
github.com/contiv/netplugin/drivers.(*OvsDriver).DeletePeerHost(0xc8201d8c40, 0xc8204180dd, 0xc, 0x2332, 0x0, 0x0)
	/opt/gopath/src/github.com/contiv/netplugin/drivers/ovsdriver.go:389 +0x217
github.com/contiv/netplugin/netplugin/plugin.(*NetPlugin).DeletePeerHost(0xc82005bb00, 0xc8204180dd, 0xc, 0x2332, 0x0, 0x0)
	/opt/gopath/src/github.com/contiv/netplugin/netplugin/plugin/netplugin.go:138 +0x64
github.com/contiv/netplugin/netplugin/cluster.peerDiscoveryLoop(0xc82005bb00, 0x7f6004a99640, 0xc82008aac0, 0xc820010d47, 0xc)
	/opt/gopath/src/github.com/contiv/netplugin/netplugin/cluster/cluster.go:212 +0x1016
created by github.com/contiv/netplugin/netplugin/cluster.Init
	/opt/gopath/src/github.com/contiv/netplugin/netplugin/cluster/cluster.go:279 +0xb3
```

details and log at https://gist.github.com/vvb/95c625a356c9eb5cd238

--- a/drivers/ovsdbDriver.go
+++ b/drivers/ovsdbDriver.go
@@ -574,7 +574,7 @@ func (d *OvsdbDriver) GetOfpPortNo(intfName string) (uint32, error) {
 	for {
 		row, err := d.ovs.Transact(ovsDataBase, selectOp)
 
-		if err == nil {
+		if err == nil && len(row) > 0 && len(row[0].Rows) > 0 {
 			value := row[0].Rows[0]["ofport"]
 			if reflect.TypeOf(value).Kind() == reflect.Float64 {
 				//retry few more time. Due to asynchronous call between

--- a/netplugin/cluster/cluster.go
+++ b/netplugin/cluster/cluster.go
@@ -185,14 +185,6 @@ func peerDiscoveryLoop(netplugin *plugin.NetPlugin, objdbClient objdb.API, local
 				if err != nil {
 					log.Errorf("Error adding node {%+v}. Err: %v", nodeInfo, err)
 				}
-				// add the node
-				err = netplugin.AddPeerHost(core.ServiceInfo{
-					HostAddr: nodeInfo.HostAddr,
-					Port:     ofnet.OFNET_AGENT_VLAN_PORT,
-				})
-				if err != nil {
-					log.Errorf("Error adding node {%+v}. Err: %v", nodeInfo, err)
-				}
 
 			} else if srvEvent.EventType == objdb.WatchServiceEventDel {
 				log.Infof("Node delete event for {%+v}", nodeInfo)
@@ -201,14 +193,6 @@ func peerDiscoveryLoop(netplugin *plugin.NetPlugin, objdbClient objdb.API, local
 				err := netplugin.DeletePeerHost(core.ServiceInfo{
 					HostAddr: nodeInfo.HostAddr,
 					Port:     ofnet.OFNET_AGENT_VXLAN_PORT,
-				})
-				if err != nil {
-					log.Errorf("Error adding node {%+v}. Err: %v", nodeInfo, err)
-				}
-				// remove the node
-				err = netplugin.DeletePeerHost(core.ServiceInfo{
-					HostAddr: nodeInfo.HostAddr,
-					Port:     ofnet.OFNET_AGENT_VLAN_PORT,
 				})
 				if err != nil {
 					log.Errorf("Error adding node {%+v}. Err: %v", nodeInfo, err)

